src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php

```diff
@@ -143,4 +146,27 @@ private function isForceEager(string $resourceClass, array $options): bool
 
         return is_bool($forceEager) ? $forceEager : $this->forceEager;
     }
+
+    private function hasFetchEagerAssociation(EntityManager $em, ClassMetadataInfo $classMetadata, &$checked = [])
```

Comment:
I don't want to to this to avoid a 2-pass recursive call in `EagerLoadingExtension`. If we want to extract this logic we need to add cache. I obviously agree that we should add a new class with the `shouldEagerLoad` logic.